### PR TITLE
feat(cli): auto-initialize git and run tambo init by default

### PR DESCRIPTION
**Pull request by akhileshrangani4**
## Summary

- `npm create tambo-app` now automatically initializes git and runs `npx tambo init` by default
- Added `--skip-git-init` and `--skip-tambo-init` flags to disable automatic setup
- Tambo init automatically skipped in non-interactive environments (CI/CD, agents)
- Improved output messages to clearly show what was done automatically vs what remains
- Updated all documentation (README, quickstart, CLI reference) to reflect new behavior
- Removed deprecated `--init-git` flag (use `--skip-git-init` to disable instead)

## Why

Reduces friction for new users by automating the most common setup steps. Previously users had to manually:
1. Initialize git repository
2. Run `npx tambo init` for API key setup

Now both happen automatically in interactive mode, while CI/CD environments automatically skip the interactive `tambo init` step. Users can still disable either behavior with flags.

## Implementation Details

**Git Init:**
- Runs by default (unless `--skip-git-init` specified)
- Creates initial commit after cloning template
- Tracks success/failure separately from tambo init

**Tambo Init:**
- Only runs in interactive mode (TTY detection + not CI)
- Uses `stdio: "inherit"` to show prompts to user
- Automatically skipped in CI/CD with clear messaging
- Can be manually skipped with `--skip-tambo-init`
- Runs without modifications - full interactive experience

**Output Logic:**
- Fixed dead code where celebratory message never showed
- Separated setup steps (git/tambo) from navigation steps (cd/npm run dev)
- Shows contextual messages: "failed", "skipped", "required for setup"
- Celebrates when both git and tambo init succeed

**Deprecated Flag Removal:**
- Removed `--init-git` flag entirely (git init now defaults to true)
- Users who want to skip it use `--skip-git-init`
- Simpler logic, no edge cases to handle

## Test Plan

**Interactive Mode:**
```bash
# Test default behavior (should run git init + tambo init)
npm create tambo-app@latest test-app-1

# Test skipping git init
npm create tambo-app@latest test-app-2 --skip-git-init

# Test skipping tambo init
npm create tambo-app@latest test-app-3 --skip-tambo-init

# Test skipping both
npm create tambo-app@latest test-app-4 --skip-git-init --skip-tambo-init
```

**Non-Interactive Mode:**
```bash
# Should auto-skip tambo init with clear message
CI=true npm create tambo-app@latest test-app-ci --template=standard
```

**Verify:**
- [ ] Git initialized with initial commit (unless skipped)
- [ ] Tambo init runs and prompts for auth (interactive only)
- [ ] Output messages match what actually happened
- [ ] CI workflow still passes (tambo init auto-skipped)
- [ ] Documentation is accurate

## Breaking Changes

None - this is backwards compatible:
- Default behavior is now opt-out instead of opt-in
- Non-interactive mode automatically adapts
- Removed deprecated `--init-git` flag (use `--skip-git-init` instead)

Fixes TAM-1187

### CODE REVIEW

File: cli/src/commands/create-app.ts [hunk 1]
```diff
@@ -270,23 +272,29 @@ export async function handleCreateApp(
...
     process.chdir(targetDir);
 
-    // Initialize new git repository if requested
-    if (options.initGit) {
+    // Initialize new git repository by default (unless skipGitInit is true)
+    // Support legacy initGit flag for backwards compatibility
+    const shouldInitGit = options.skipGitInit ? false : true;
+    let gitInitSucceeded = false;
+
+    if (shouldInitGit) {
```
[the remark this answers is not in the document]
Comment (akhileshrangani4) [reply cli/src/commands/create-app.ts:278]: did this.

**Review by lachieh — APPROVED** [review]
